tests/src/test/resources/auth/application.yml

```diff
@@ -25,5 +25,7 @@ quarkus:
     category:
       "io.quarkus.vertx.core.runtime":
         level: DEBUG
+      "org.eclipse.hono":
+        level: INFO
```

Comment:
The links from modules to tests/demo-certs are just in .env and in some pom.xml optional profiles. The modules could be build without dependencies. The images doesn't contain none of .env/test/demo-certs. FMPOV there is no hard linking and no real issues.
Having maven and quarkus specifics the only think I could imaging to remove duplication is to remove some yaml files from resources and replace with .env. However there are differences (e.g. the need of .env to solve port collision) and this could become very complicated task (if possible).
I think that current approach is the the fastest and simplest (though a little dirty) and is sufficient to provide tools for easier development  with as little as possible changes of the current project on the price of little maintenance overhead